pkg/actuators/machine/actuator.go

```diff
@@ -0,0 +1,221 @@
+	providerSpec, err := ovirtconfigv1.ProviderSpecFromRawExtension(machine.Spec.ProviderSpec.Value)
+	if err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.InvalidMachineConfiguration(
+			"Cannot unmarshal machineProviderSpec field: %v", err))
+	}
+
+	connection, err := actuator.getConnection()
+	if err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.InvalidMachineConfiguration(
+			"failed to create connection to oVirt API: %v", err))
+	}
+
+	ovirtClient := ovirtC.NewOvirtClient(connection)
+	mScope := newMachineScope(ctx, ovirtClient, actuator.client, actuator.machinesClient, machine, providerSpec)
+
+	if err := mScope.create(); err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.CreateMachine(
+			"Error creating Machine %v", err))
+	}
+
+	if err := mScope.patchMachine(ctx, conditionSuccess()); err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.CreateMachine(
+			"Error patching Machine %v", err))
+		return err
+	}
+	actuator.EventRecorder.Eventf(machine, corev1.EventTypeNormal, "Created", "Updated Machine %v", machine.Name)
+	return nil
+}
+
+// Exists determines if the given machine currently exists.
+// A machine which is not terminated is considered as existing.
+func (actuator *OvirtActuator) Exists(ctx context.Context, machine *machinev1.Machine) (bool, error) {
+	klog.Infof("Checking machine %v exists.\n", machine.Name)
+	connection, err := actuator.getConnection()
+	if err != nil {
+		return false, fmt.Errorf("failed to create connection to oVirt API")
+	}
+	ovirtClient := ovirtC.NewOvirtClient(connection)
+	if err != nil {
+		return false, err
+	}
+	mScope := newMachineScope(ctx, ovirtClient, actuator.client, actuator.machinesClient, machine, nil)
+	return mScope.exists()
+}
+
+// Update attempts to sync machine state with an existing instance.
+// Updating provider fields is not supported, a new machine should be created instead
+func (actuator *OvirtActuator) Update(ctx context.Context, machine *machinev1.Machine) error {
+	// eager update
+	providerSpec, err := ovirtconfigv1.ProviderSpecFromRawExtension(machine.Spec.ProviderSpec.Value)
+	if err != nil {
+		return actuator.handleMachineError(machine, "Update", apierrors.InvalidMachineConfiguration(
+			"Cannot unmarshal machineProviderSpec field: %v", err))
+	}
+
+	connection, err := actuator.getConnection()
+	if err != nil {
+		return actuator.handleMachineError(machine, "Update", apierrors.UpdateMachine(
+			"failed to create connection to oVirt API %v", err))
+	}
+
+	ovirtClient := ovirtC.NewOvirtClient(connection)
+	mScope := newMachineScope(ctx, ovirtClient, actuator.client, actuator.machinesClient, machine, providerSpec)
+	if err := mScope.patchMachine(ctx, conditionSuccess()); err != nil {
+		return actuator.handleMachineError(machine, "Update", apierrors.UpdateMachine(
+			"Error patching Machine %v", err))
+		return err
```

Comment:
This return seems to be ineffectual. Please start using `golangci-lint` or `go vet` to detect code faults like this. (The former is more powerful.)